--- configure.in	2008-08-08 13:57:57.000000000 -0700
+++ configure.tmp	2008-12-29 08:47:41.000000000 -0800
@@ -467,6 +467,17 @@
        eolstr="\\n"
        proc_mutex_is_global=1
        OBJECTS_PLATFORM='$(OBJECTS_win32)'
+       if test "$cross_compiling" = yes; then
+           ac_cv_file__dev_zero=no
+           ac_cv_func_setpgrp_void=no
+           apr_cv_tcp_nodelay_with_cork=no
+           case $host in
+               *mingw32*)
+                   ac_cv_sizeof_ssize_t=4
+                   ac_cv_sizeof_size_t=4
+                   ;;
+           esac
+       fi
        ;;
    *cygwin*)
        OSDIR="unix"
